libimagstore/src/store.rs

```
@@ -147,7 +147,17 @@ impl Store {
 
     /// Delete an entry
     pub fn delete(&self, id: StoreId) -> Result<()> {
-        unimplemented!();
+        let mut entries_lock = self.entries.write();
+        let mut entries = entries_lock.unwrap();
+
+        // if the entry is currently modified by the user, we cannot drop it
+        if entries.get(&id).map( |e| e.is_borrowed() ).unwrap_or(false) {
```

You're hiding an error here. If the `get()` call fails due to the idea not beeing in the store, the error is still `IdLocked`.

Also: Closures are without trailing whitespace: `map(|e| e.is_borrowed())`.